internal/controllers/gateway/route_utils_test.go

```diff
@@ -527,6 +547,550 @@ func Test_getSupportedGatewayForRoute(t *testing.T) {
+						},
+						Spec: gatewayv1beta1.GatewaySpec{
+							GatewayClassName: "test-gatewayclass",
+							Listeners: []gatewayv1beta1.Listener{
+								{
+									Name:     gatewayv1beta1.SectionName("http"),
+									Protocol: gatewayv1beta1.HTTPProtocolType,
+									Port:     gatewayv1beta1.PortNumber(80),
+									AllowedRoutes: &gatewayv1beta1.AllowedRoutes{
+										Namespaces: &gatewayv1beta1.RouteNamespaces{
+											From: addressOf(gatewayv1beta1.NamespacesFromSame),
+										},
+									},
+									Hostname: addressOf(gatewayv1beta1.Hostname("hostname.com")),
+								},
+							},
+						},
+						Status: gatewayv1beta1.GatewayStatus{
+							Listeners: []gatewayv1beta1.ListenerStatus{
+								{
+									Name: gatewayv1beta1.SectionName("http"),
+									Conditions: []metav1.Condition{
+										{
+											Type:   "Ready",
+											Status: metav1.ConditionTrue,
+										},
+									},
+									SupportedKinds: []gatewayv1beta1.RouteGroupKind{
+										{
+											Group: addressOf(gatewayv1beta1.Group(gatewayv1beta1.GroupVersion.Group)),
+											Kind:  gatewayv1beta1.Kind("HTTPRoute"),
+										},
+									},
+								},
+							},
+						},
+					},
+					&GatewayClass{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test-gatewayclass",
+						},
+						Spec: gatewayv1beta1.GatewayClassSpec{
+							ControllerName: gatewayv1beta1.GatewayController("konghq.com/kic-gateway-controller"),
+						},
+					},
+					&corev1.Namespace{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test-namespace",
+						},
+					},
+				},
+				expected: []expected{
+					{
+						gateway: types.NamespacedName{
+							Name:      "test-gateway",
+							Namespace: "test-namespace",
+						},
+						listenerName: "",
+						condition: metav1.Condition{
+							Type:   string(gatewayv1beta1.RouteConditionAccepted),
+							Status: metav1.ConditionFalse,
+							Reason: string(gatewayv1beta1.RouteReasonNoMatchingListenerHostname),
+						},
+					},
+				},
+			},
+		}
+
+		for _, tt := range tests {
+			tt := tt
+			t.Run(tt.name, func(t *testing.T) {
+				fakeClient := fakeclient.
+					NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithObjects(tt.objects...).
+					Build()
+
+				got, err := getSupportedGatewayForRoute(context.Background(), fakeClient, tt.route)
+				if tt.wantErr {
+					require.Error(t, err)
+				} else {
+					require.NoError(t, err)
+					require.Len(t, got, len(tt.expected))
+
+					for i := range got {
+						assert.Equalf(t, tt.expected[i].gateway.Namespace, got[i].gateway.Namespace, "gateway namespace #%d", i)
+						assert.Equalf(t, tt.expected[i].gateway.Name, got[i].gateway.Name, "gateway name #%d", i)
+						assert.Equalf(t, tt.expected[i].listenerName, got[i].listenerName, "listenerName #%d", i)
+						assert.Equalf(t, tt.expected[i].condition, got[i].condition, "condition #%d", i)
+					}
+				}
+			})
+		}
+	})
+
+	t.Run("TCPRoute", func(t *testing.T) {
```

Comment:
I think the tests should contain at least one accepting case for each type of route.

Reply:
Since only HTTPRoute is supported as of now (and the remaining route types are supported on a best effort basis: mostly because they are still in alpha in Gateway API) I feel what's proposed in this PR (some, minor coverage for `TCPRoute`) to be enough. WDYT/